blackjack/core.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python
+
+import enum
+import random
+from typing import List, Optional
+
+
+class Win(enum.Enum):
+    Push = enum.auto()
+    Player_bj = enum.auto()
+    House_bj = enum.auto()
+    Player_bust = enum.auto()
+    House_bust = enum.auto()
+    Player_win = enum.auto()
+    House_win = enum.auto()
+
+
+class Play(enum.Enum):
+    Hit = enum.auto()
+    Stay = enum.auto()
+    # TODO Add Split
+
+
+class Suit(enum.Enum):
+    Clubs = enum.auto()
+    Hearts = enum.auto()
+    Spades = enum.auto()
+    Diamonds = enum.auto()
+
+
+class Rank(enum.Enum):
+    Ace = enum.auto()
+    Two = enum.auto()
+    Three = enum.auto()
+    Four = enum.auto()
+    Five = enum.auto()
+    Six = enum.auto()
+    Seven = enum.auto()
+    Eight = enum.auto()
+    Nine = enum.auto()
+    Ten = enum.auto()
+    Jack = enum.auto()
+    Queen = enum.auto()
+    King = enum.auto()
+
+
+class Card:
+    def __init__(self, rank: Rank, suit: Suit):
+        self.rank = rank
+        self.suit = suit
+
+    def __repr__(self):
+        return f"<Card({self.rank.name}, {self.suit.name})>"
+
+
+class GameState:
+    def __init__(self, deck: List[Card], player_hand: List[Card], dealer_hand: List[Card]):
+        self.deck = deck
+        self.player_hand = player_hand
+        self.dealer_hand = dealer_hand
+
+    def deal(self):
+        """Removes and returns first card of the deck."""
+        return self.deck.pop(0)
+
+
+def make_deck():
+    deck = []
+    for suit in Suit:
+        for rank in Rank:
+            card = Card(rank, suit)
+            deck.append(card)
+    return deck
+
+
+def house_hit(state: GameState):
+    while hand_total(state.dealer_hand) <= 17:
+        state.dealer_hand.append(state.deal())
+
+
+def hit_player(state: GameState):
+    state.player_hand.append(state.deal())
+
+
+def hit(hit_response, state: GameState):
+    if hit_response == Play.Hit:
+        hit_player(state)
+        return
+    if hit_response == Play.Stay:
+        house_hit(state)
+        return
+    raise ValueError(f"Invalid Play: {hit_response}")
+
+
+def hand_total(hand):
+    total = 0
+    ace_count = 0
+    for card in hand:
+        r = card.rank
+        total += card_point(card)
+        if r == Rank.Ace:
+            ace_count += 1
+        while ace_count > 0 and total > 21:
+            ace_count -= 1
+            total -= 10
+    return total
+
+
+def initial_deal(state):
+    state.player_hand.append(state.deal())
+    state.player_hand.append(state.deal())
+    state.dealer_hand.append(state.deal())
+    state.dealer_hand.append(state.deal())
```

Comment:
This is not right. You should deal to the player and _then_ the dealer. When you play a card game you go around the table when you deal cards.